### ISSUE TITLE
Spectral Workbench goals

### ISSUE THREAD

**Issue by jywarren** [publiclab/spectral-workbench.js#159]
## High priority

* modernize getUserMedia integration
* configurable capture resolutions
* replace spectral-workbench capture with the modularized version, as Editor is included in plots2
* develop a stand-alone program from viewer and capture modules
* develop test cases for capture program, such as doing a calibration, detecting overexposure, potentially using pre-recorded video
* create documentation for setup and initialization of capture interface

## Nice to have

* ability to save and view spectra offline, by pointing viewer at a standard local JSON file
* requirements list for current capture UI
* new UI (react?) that fulfills the above (including rotate, flip, etc)
* make a static/standalone demo page for ability to upload an image and calibrate and graph it, extract JSON

**Comment by jywarren** [comment]
* https://publiclab.org/wiki/spectral-workbench-snapshots
* https://publiclab.org/wiki/spectral-workbench-calibration
* https://publiclab.org/wiki/spectral-workbench-tools
* https://publiclab.org/wiki/spectral-workbench-operations

(working on fixing these videos today, sorry!)

**Comment by jywarren** [comment]
Just chiming in that the above tutorial videos are all fixed now! cc @alaxalves -- sorry it took a while!

**Comment by alaxalves** [comment]
OMG @jywarren I was totally unaware of this JS package!! The more I navigate through PublicLab the more I find new stuff. Since this is spectral-related maybe I can find some time in my GSOC schedule to contribute to this too. Thanks for taking time on those tutorials. :smile:

**Comment by jywarren** [comment]
:-)

that would be amazing! thanks, Alax!

Just noting that i got this running on Node 8 (https://github.com/publiclab/spectral-workbench.js/pull/165) but node 10 is blocked by a globals usage that may take significant refactoring: #162  -- and that we seem to be limited on how far we can upgrade d3 because of our use of nvd3 wrapper library (read more in #155).

However, i think a lot of the dependabot upgrades will start working more now that we are on Node 8.

**Comment by jywarren** [comment]
Many of these may be complete - let's update our list and close relevant issues, but only if they are very close to going live on SpectralWorkbench.org, maybe? Or just note if there are any remaining steps before they go live (i.e. their status):

**Comment by jywarren** [comment]
Updated this after the huge project of this summer by @waridrox !!! Great work!! https://github.com/publiclab/spectral-workbench/issues/645
